commands/files/serialfile.go

```diff
@@ -84,10 +84,25 @@ func (f *serialFile) NextFile() (File, error) {
 	// open the next file
 	fileName := fp.Join(f.name, stat.Name())
 	filePath := fp.Join(f.path, stat.Name())
+	st, err := os.Lstat(filePath)
```

Comment:
I think that we only care to handle symlinks like that here, where the primary use case is `ipfs add -r`.